runtime/weak.c

```diff
@@ -358,20 +358,28 @@ CAMLprim value caml_ephe_get_data (value ar)
   return optionalize(caml_ephemeron_get_data(ar, &data), &data);
 }
 
-
-Caml_inline void copy_value(value src, value dst)
-{
-  if (Tag_val (src) < No_scan_tag){
-    mlsize_t i;
-    for (i = 0; i < Wosize_val (src); i++){
-      value f = Field (src, i);
-      if (caml_gc_phase == Phase_mark && Must_be_Marked_during_mark(f)){
-        caml_darken (f, NULL);
-      }
-      caml_modify (&Field (dst, i), f);
+static void copy_value(value src, value dst)
+{
+  mlsize_t sz, i;
+  sz = Wosize_val(src);
+  if (Tag_val (src) >= No_scan_tag) {
+    /* Direct copy */
+    memcpy (Bp_val (dst), Bp_val (src), Bsize_wsize (sz));
+    return;
+  }
+  i = 0;
+  if (Tag_val (src) == Closure_tag) {
+    /* Direct copy of the code pointers and closure info fields */
+    i = Start_env_closinfo(Closinfo_val(src));
+    memcpy (Bp_val (dst), Bp_val (src), Bsize_wsize (i));
```

Comment:
Why using memcpy for 1 word and not simply an assignment?

Reply:
This code copies `i` words, where `i` is the size of the non-environment part of the closure.